### ISSUE TITLE
Need some help with commands

### ISSUE THREAD

**Issue by andesse** [simeonovp/node-red-contrib-tuya-devices#9]
I have a Diffuser that used to work with the other Tuya contrib, but i don't get a connection anymore. With this contrib it works, but I am not able to use my previous commands. ![image](https://github.com/user-attachments/assets/d7ca3ce4-f48f-4535-9e74-ff77009ef150)

When I use "get status of dps" I get this back 
![image](https://github.com/user-attachments/assets/0b9199c0-9b83-4a7f-b977-af9598b5d51a)

How do I need to set the commands that it works with this contrib? Can someone give me a hint?

**Comment by andesse** [comment]
![image](https://github.com/user-attachments/assets/ef30c528-8f38-4d32-880f-c5b6ac152b49)

**Comment by simeonovp** [comment]
@andesse You can update to the current version (1.3.1) and try the new feature "Multi Outputs". Open device GUI, select "Datapoint":"All", "Outputs":"Multi outputs". The first output gets control content and raw messages. All other outputs are for the available DPs (holt the mouse over the output to see the tooltip). In addition you can inject messages with topic "dispatchCommands" or "getDataModel" to the device node to get additional infos on the data (first) output. You can also directly attach dashboard nodes (depends oto the DP data type) to the outputs and to the input of the device node. The topic in the dashboard node must be the code of the DP (see its model)

**Comment by andesse** [comment]
Thanks for the feedback, still not working. The dropdown doesn't show me anything. ![image](https://github.com/user-attachments/assets/1379d64a-bcdc-4c08-bc9f-737e5e1d638a)
Get Data Model gives me an empty payload.

Dispatch gives me this ![image](https://github.com/user-attachments/assets/ab9f2f5e-8afa-4a8a-b592-6014b7f68070)

I actually just need to know how to format my inject data. Mine as shown on the first pictures is not working and shows the error on the additional picture

**Comment by simeonovp** [comment]
> Thanks for the feedback, still not working. The dropdown doesn't show me anything. ![image](https://private-user-images.githubusercontent.com/76150626/373958919-1379d64a-bcdc-4c08-bc9f-737e5e1d638a.jpeg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjgyMzA0OTMsIm5iZiI6MTcyODIzMDE5MywicGF0aCI6Ii83NjE1MDYyNi8zNzM5NTg5MTktMTM3OWQ2NGEtYmNkYy00YzA4LWJjOWYtNzM3ZTVlMWQ2MzhhLmpwZWc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMDA2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTAwNlQxNTU2MzNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0yMTYzZWVkMzc4NGU3M2E3Yjg3ZmIwNzAzMjA0N2E2MTRkNDc5OGEwYjljZjg0MGQ2Y2E5MTQ4MmI1MDMzYzExJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.AwrEZaKGnNZMydRDhTQ35W-W55shxP0sKdbnoYwRhTc) Get Data Model gives me an empty payload.
> 
> Dispatch gives me this ![image](https://private-user-images.githubusercontent.com/76150626/373958983-ab9f2f5e-8afa-4a8a-b592-6014b7f68070.jpeg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjgyMzA0OTMsIm5iZiI6MTcyODIzMDE5MywicGF0aCI6Ii83NjE1MDYyNi8zNzM5NTg5ODMtYWI5ZjJmNWUtOGFmYS00YThhLWI1OTItNjAxNGI3ZjY4MDcwLmpwZWc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMDA2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTAwNlQxNTU2MzNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1kYzExODk1NGU4NTg4NGMzOTY3YmEyYmU1MzQxMGQzYmI4NmM3NTFjZTg3YTEyYTYyN2I0MzczMjVjNjcxOGIwJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.Lp2YPchXbpQsiRbOwnHEcJHU6hMcvNDglTtXXREaoPc)
> 
> I actually just need to know how to format my inject data. Mine as shown on the first pictures is not working and shows the error on the additional picture
> 
> I don't care for outputs, I jut need to run the device in different modes

It seam so as you not synchronize you local cache with the cloud. The deviceand model informations comes primary from the cloud and will be cached localy for future usage. You must once press the Refresh button in the manager GUI.

**Comment by andesse** [comment]
Okey, I'll try that. I used my devices locally before with internet access blocked.

**Comment by andesse** [comment]
Every interaction with the Manager Node Crashs my Node-Red. 😅
My credentials from the Tuya developer site are correct. 
I will investigate further.

**Comment by corckie** [comment]
hey @simeonovp I have the same question, what is the format to change the node status?

Here is the response from my lightbulb:

![image](https://github.com/user-attachments/assets/8a7eb3f4-acc2-4497-8a38-b480094758aa)

[20] is lightbulb status - true or false. How do I set this variable?

**Comment by corckie** [comment]
@andesse 

In case you need some more help:

One lightbulb that I use is Wi-Fi connected and it needs this payload:

![Image](https://github.com/user-attachments/assets/5b26ac40-b4a2-4777-8bcf-ec7e057bc9f4)

msg.payload = (change payload type to json!) {   "20": true } . True = turn on, false = turn off.
msg.dispatchCommands = set (payload type - string)

20 - it's DPS number. So 20 for state, 21 for brightness, 22 for temperature etc.

When it comes to a Zigbee Bridge, in my case, it works differently. I send a command to a Zigbee device connected to a bridge, BUT this time the payload is easier.

![Image](https://github.com/user-attachments/assets/7cce7498-5746-4d1e-96fa-5de715f30e49)

msg.payload = {"1":true} (change type to JSON!)

1 is DPS for switch on/of,

3 is for brightness. In my case it's between 12 (1% brightness) and 1000 (100% brightness).

![Image](https://github.com/user-attachments/assets/898d779a-bc3b-43aa-9e90-523021003602)

So it looks like that depending on the device and protocol it might work differently. Wifi lightbulb has different methods than a Zigbee lightbulb.

**Comment by vampywiz17** [comment]
@corckie 

Yes, the tuya implementation is different all devices... these things are write the actual OEM provider. For example tuya are accept RGB and HSV  model... one OEM use one, other is the other one.

**Comment by simeonovp** [comment]
You can set the value in the payload and a DPs code, e.g. "switch_1" (see tooltip of the output) as topic .
